src/rez/utils/pip.py

```diff
@@ -0,0 +1,705 @@
+
+    # create cased names lookup
+    name_mapping = dict((x.lower(), x) for x in (name_casings or []))
+
+    # requirements such as platform, arch, os, and python
+    sys_requires = set(["python"])
+
+    # assume package is platform- and arch- specific if it isn't pure python
+    is_pure_python = is_pure_python_package(installed_dist)
+    if not is_pure_python:
+        sys_requires.update(["platform", "arch"])
+
+    # Note: This is supposed to give a requirements list that has already been
+    # filtered down based on the extras requested at install time, and on any
+    # environment markers present. However, this is not working in distlib. The
+    # package gets assigned a LegacyMetadata metadata object, and in that code
+    # path, this filtering doesn't happen.
+    #
+    # See: vendor/distlib/metadata.py#line-892
+    #
+    requires = installed_dist.run_requires
+
+    # filter requirements
+    for req_ in requires:
+        reqs = normalize_requirement(req_)
+
+        for req in reqs:
+            # skip if env marker is present and doesn't evaluate
+            if req.marker and not req.marker.evaluate():
+                continue
+
+            # skip if req is conditional on extras that weren't requested
+            if req.conditional_extras and not \
+                    (set(installed_dist.extras or []) & set(req.conditional_extras)):
+                continue
+
+            if req.conditional_extras:
+                print_warning(
+                    "Skipping requirement %r - conditional requirements are "
+                    "not yet supported", str(req)
+                )
+                continue
+
+            # Inspect marker(s) to see if this requirement should be varianted.
+            # Markers may also cause other system requirements to be added to
+            # the variant.
+            #
+            to_variant = False
+
+            if req.marker:
+                marker_reqs = get_marker_sys_requirements(str(req.marker))
+
+                if marker_reqs:
+                    sys_requires.update(marker_reqs)
+                    to_variant = True
+
+            # remap the requirement name
+            remapped = name_mapping.get(req.name.lower())
+            if remapped:
+                req.name = remapped
+
+            # convert the requirement to rez equivalent
+            rez_req = str(packaging_req_to_rez_req(req))
+
+            if to_variant:
+                result_variant_requires.append(rez_req)
+            else:
+                result_requires.append(rez_req)
+
+    # prefix variant with system requirements
+    sys_variant_requires = []
+
+    if "platform" in sys_requires:
+        sys_variant_requires.append("platform-%s" % _system.platform)
+
+    if "arch" in sys_requires:
+        sys_variant_requires.append("arch-%s" % _system.arch)
+
+    if "os" in sys_requires:
+        sys_variant_requires.append("os-%s" % _system.os)
+
+    if "python" in sys_requires:
```

Comment:
Is this if necessary?

Reply:
I think it's cleaner to have it here, even though python is always a requirement. Then all the code that deals with converting a simple list of system requirements, into a more specific list, is all in the one place. Especially as later, it may get more complex and may not just default to python-MAJOR.MINOR.